riocli/device/delete.py

```diff
@@ -27,41 +36,148 @@
-        handle_device_delete_error(client.delete_device(device_id=device_guid))
-        spinner.text = click.style('Device deleted successfully', fg=Colors.GREEN)
-        spinner.green.ok(Symbols.SUCCESS)
+        result = Queue()
+        func = functools.partial(_delete_deivce, client, result)
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            executor.map(func, devices)
+
+        result = sorted(list(result.queue), key=lambda x: x[0])
+
+        data, fg, statuses = [], Colors.GREEN, []
+        success_count, failed_count = 0, 0
+
+        for name, response in result:
+            if response.status_code and response.status_code < 400:
+                fg = Colors.GREEN
+                icon = Symbols.SUCCESS
+                success_count += 1
+                msg = ''
+            else:
+                fg = Colors.RED
+                icon = Symbols.ERROR
+                failed_count += 1
+                msg = get_error_message(response, name)
+
+            data.append([
+                click.style(name, fg),
+                click.style(icon, fg),
+                click.style(msg, fg)
+            ])
+
+        with spinner.hidden():
+            tabulate_data(data, headers=['Name', 'Status', 'Message'])
+
+        spinner.write('')
+
+        if failed_count == 0 and success_count == len(devices):
+            spinner_text = click.style('All devices deleted successfully.', Colors.GREEN)
+            spinner_char = click.style(Symbols.SUCCESS, Colors.GREEN)
+        elif success_count == 0 and failed_count == len(devices):
+            spinner_text = click.style('Failed to delete devices', Colors.YELLOW)
+            spinner_char = click.style(Symbols.WARNING, Colors.YELLOW)
+        else:
+            spinner_text = click.style(
+                '{}/{} devices deleted successfully'.format(success_count, len(devices)), Colors.YELLOW)
+            spinner_char = click.style(Symbols.WARNING, Colors.YELLOW)
+
+        spinner.text = spinner_text
+        spinner.ok(spinner_char)
     except Exception as e:
-        spinner.text = click.style('Failed to delete device: {}'.format(e), fg=Colors.RED)
+        spinner.text = click.style(
+            'Failed to delete devices: {}'.format(e), Colors.RED)
         spinner.red.fail(Symbols.ERROR)
         raise SystemExit(1) from e
 
 
-def handle_device_delete_error(response: Response):
-    if response.status_code < 400:
-        return
+def fetch_devices(
+        client: Client,
+        device_name_or_regex: str,
+        delete_all: bool,
+) -> List[Device]:
+    devices = client.get_all_devices()
+    result = []
+    for device in devices:
+        if (delete_all or device.name == device_name_or_regex or
+                (device_name_or_regex not in device.name and
+                 re.search(device_name_or_regex, device.name)) or
+                device_name_or_regex == device.uuid):
+            result.append(device)
+
+    return result
```

Comment:
Let's move this to `riocli/device/util.py` such that we reuse it at other places as well.